CMakeLists.txt

```diff
@@ -2030,6 +2032,27 @@ if(WIN32)
   target_include_directories(mixxx PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}")
 endif()
 
+# ASIO (included in Ableton Link) requires atomic on clang to link `std::__atomic_base<unsigned int>::is_lock_free() const'
+include(CheckAtomic)
+if(HAVE_CXX_ATOMICS_WITH_LIB OR HAVE_CXX_ATOMICS64_WITH_LIB)
+  message(STATUS "Linking libatomic is needed for atomic support on this platform")
+  target_link_libraries(mixxx-lib PUBLIC atomic)
+endif()
+
+# Ableton Link
+find_package(AbletonLink NAMES AbletonLink ableton ableton-link ableton-link-dev)
+# Note Debian Bug#993999 - Wrong CMake include path: https://salsa.debian.org/multimedia-team/ableton-link/-/commit/047f75abeeb6494256cb8d498995c17afd2a17e8
+
+if (AbletonLink_FOUND)
+  message(STATUS "Using existing system installation of Ableton Link")
+  target_link_libraries(mixxx-lib PRIVATE Ableton::Link)
+else()
+  message(STATUS "Using internal Ableton Link")
+  include(${CMAKE_CURRENT_SOURCE_DIR}/lib/ableton-link/share/ableton/AbletonLinkConfig.cmake)
+  target_link_libraries(mixxx-lib PUBLIC Ableton::Link)
+  target_include_directories(mixxx-lib SYSTEM PUBLIC lib/ableton-link/include/)
+endif()
```

Comment:
fyi this doesn't work. Did you forget to include the `AbletonLinkConfig.cmake` or whats going on here? Do we even want to fendor this? availability is still quite sparse on linux unfortunately. https://repology.org/project/ableton-link/versions

Reply:
I don't understand the point. It builds on all three  platforms including Ubuntu. If the package is not available on unsuported third party Linux distributions, it's the job of these distributions maintainers to create such a package.

Reply:
Okay thanks for clarifying. I was under the impression that we would fallback to a vendored version of ableton-link if it was not available on the system. Similar to how we do it with libshout-idjc or portaudio or any other library in our `lib/` folder. That usually works, except in this case the files for that are obviously missing from the lib folder and thus CMake just plain fails, which is definitely a bug. So either we fix the internal fallback, or remove the feature entirely.